packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -155,7 +156,81 @@ export async function forceLightningLwcPreview(sourceUri: vscode.Uri) {
     return;
   }
 
-  await selectPlatformAndExecute(startTime, componentName);
```

Comment:
Rather than having a giant `selectPlatformAndExecute` function I've broken it down and refactored it into multiple functions (each responsible for doing part of the overall action)